Remove explicit DNS flush on darwin.

thallgren opened the pull request:
## Description

Updating files under `/etc/resolver` is enough to flush the DNS cache
on macOS. There's no need to flush it explicitly. Actually, doing so
makes the workstations DNS somewhat unstable since requests made during
the flush can be interrupted and return failures.

## Checklist

 - [x] I made sure to update `./CHANGELOG.md`.
 - [x] My change is adequately tested.

Commit:
Remove explicit DNS flush on darwin.

Updating files under `/etc/resolver` is enough to flush the DNS cache
on macOS. There's no need to flush it explicitly. Actually, doing so
makes the workstations DNS somewhat unstable since requests made during
the flush can be interrupted and return failures.

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

josecv reviewed (approved):
looks good and tracks with what we said on wednesday but i'm a little concerned about the failing tests; approving bc i assume it's just a flake though

Commit:
Remove optimization of DNS update.

Just picking the last update of search paths might result in
an attempt to set the exact same search paths as the previous
setting. That in turn is a no-op. This is bad if the paths were
changed in between since it actually disables the DNS-flush
on some systems (notably darwin).

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

thallgren commented:
Closing this in favor of #2210.
